logger/include/logger.hpp

```diff
@@ -0,0 +1,57 @@
+#pragma once
+
+#include <filesystem>
+#include <iostream>
+#include <source_location>
+#include <sstream>
+
+enum class Loglevel
+{
+    INFO,
+    CRITICAL
+};
+
+class Logger
+{
+  public:
+    Loglevel level;
+    std::stringstream logmessage;
+    /**
+     * @brief checks the loglevel of the message
+     * and appends the additional information of the
+     * message like file name, function name and line number to the buffer
+     *
+     * @param[in]- log level of the message
+     * @param[in]- object of source_location class
+     */
+
+    Logger& checkloglevel(Loglevel level, const std::source_location& location =
+                                              std::source_location::current())
+    {
+        if (Loglevel::INFO == level)
+        {
+            logmessage << "File: "
+                       << std::filesystem::path(location.file_name())
+                              .filename()
+                              .string()
+                       << " Function name: " << location.function_name()
+                       << " Line no: " << location.line() << " ";
+        }
+        return *this;
+    }
+    /**
+     * @brief Operator overloading function to append message to the buffer
+     *
+     * param[in]- Information to be appended to the buffer
+     */
+
```

Comment:
No empty lines between doxygen and code.

Reply:
The data type of message can be int, char, or string. so to keep it generic I used template